utilities/workers/worker_manager.py

```diff
@@ -2,38 +2,203 @@
+                "Worker count requested is less than or equal to zero, no workers were created",
+                frame,
+            )
+            return False, None
+
+        return True, WorkerProperties(
+            cls.__create_key,
+            count,
+            target,
+            work_arguments,
+            input_queues,
+            output_queues,
+            controller,
+        )
+
+    def __init__(
+        self,
+        class_private_create_key: object,
+        count: int,
+        target: "(...) -> object",  # type: ignore
+        work_arguments: "tuple",
+        input_queues: "list[queue_proxy_wrapper.QueueProxyWrapper]",
+        output_queues: "list[queue_proxy_wrapper.QueueProxyWrapper]",
+        controller: worker_controller.WorkerController,
+    ) -> None:
+        """
+        Private constructor, use create() method.
+        """
+        assert class_private_create_key is WorkerProperties.__create_key, "Use create() method"
+
+        self.__count = count
+        self.__target = target
+        self.__work_arguments = work_arguments
+        self.__input_queues = input_queues
+        self.__output_queues = output_queues
+        self.__controller = controller
+
+    def get_worker_arguments(self) -> "tuple":
+        """
+        Concatenates the worker properties into a tuple.
+
+        Returns the worker properties as a tuple.
+        """
+        return (
+            self.__work_arguments
+            + tuple(self.__input_queues)
+            + tuple(self.__output_queues)
+            + (self.__controller,)
+        )
+
+    def get_worker_count(self) -> int:
+        """
+        Returns the worker count.
+        """
+        return self.__count
+
+    def get_worker_target(self) -> "(...) -> object":  # type: ignore
+        """
+        Returns the worker target.
+        """
+        return self.__target
+
 
 class WorkerManager:
     """
     For interprocess communication from main to worker.
     Contains exit and pause requests.
     """
 
-    def __init__(self, workers: "list[mp.Process] | None" = None) -> None:
+    __create_key = object()
+
+    @classmethod
+    def create(
+        cls,
+        worker_properties: WorkerProperties,
+    ) -> "tuple[bool, WorkerManager]":
         """
-        Constructor creates internal queue and semaphore.
+        Create identical workers and append them to a workers list.
+
+        worker_properties: Worker properties.
+
+        Returns whether the workers were able to be created and the Worker Manager.
         """
-        self.__workers = [] if workers is None else workers
+        result, worker_manager_logger = logger.Logger.create("worker_manager")
+        if not result:
+            print("Error creating worker manager logger")
+            return False, None
+
+        frame = inspect.currentframe()
+        worker_manager_logger.info("worker manager logger initialized", frame)
+
+        if worker_properties.get_worker_count() <= 0:
```

Comment:
Put the worker count into a local variable so that this method doesn't have to be repeatedly called:
```py
worker_count = worker_properties.get_worker_count()
if worker_count <= 0:
...
for _ in range(0, worker_count):
...
```